src/labone/core/shf_vector_data.py

```diff
@@ -530,57 +273,51 @@ def _deserialize_shf_scope_vector(
     )
     data_real = data[::2]
     data_imag = data[1::2]
-    return data_real + 1j * data_imag, extra_header
+    return ShfScopeVectorData(
+        vector=data_real + 1j * data_imag,
+        properties=extra_header,
+    )
 
 
-def _deserialize_shf_demodulator_vector(
-    *,
+@dataclass
+class ShfGeneratorWaveformVectorData:
+    """SHF generator waveform sample data."""
+
+    complex: np.ndarray
+
+
+def _deserialize_shf_waveform_vector(
     raw_data: bytes,
-    extra_header_info: int,
-    header_length: int,
-) -> tuple[SHFDemodSample, ShfDemodulatorVectorExtraHeader | None]:
+) -> ShfGeneratorWaveformVectorData:
     """Deserialize the vector data for waveform vectors.
 
+    Copy of CoreShfNodeData.cpp
```

Comment:
Is this necessary in the docstring or would a comment suffice, or at all?

Reply:
I removed the comment entirely. There is not use for the user